### DISCUSSION TITLE
STL import fails

### DISCUSSION THREAD

**Discussion by tim79mc** [OpenModelica/OpenModelica#9373] [General]
Hi-  new installation of OpenModelica 1.19.0 with 4.0.0 modelica library.  I created a multibody bodypart and added "file://c:/xxxxxx/file.stl" to the shapeType field.  The STL is ASCII.  The simulation runs, but when the plotting windows appears:

20:47:25 Scripting Error
Error reading file c:/xxxxxx/file.stl: read error (Could not find plugin to read objects from file "c:/xxxxxx/file.stl".) 

I scoured the web for information on plugins, STL imports, 3d visualizer - I uninstalled and reinstalled, looked through the docs and menus for answers...  Nothing.

Seems this should just work out of the box.  What am I missing?

**Comment by anotheruserofgithub** [comment]
To load the STL file, OSG's `ReaderWriterSTL` plugin is used: https://github.com/openscenegraph/OpenSceneGraph/blob/OpenSceneGraph-3.2/src/osgPlugins/stl/ReaderWriterSTL.cpp#L344

Could you tell the OpenSceneGraph version you are using?

**Comment by anotheruserofgithub** [comment]
@tim79mc
Could you please try to simulate [the attached `stl` model](https://github.com/OpenModelica/OpenModelica/files/10664594/stl.zip)? It should display a red dice. If not, please report the errors.

**Comment by anotheruserofgithub** [comment]
@tim79mc This issue has been solved. Thank you for reporting it! The fix is already available in the [nightly builds](https://build.openmodelica.org/omc/builds/windows/nightly-builds/) and will be shipped in the forthcoming [release 1.21](https://github.com/OpenModelica/OpenModelica/milestone/41).

**Comment by anotheruserofgithub** [comment]
@adeas31 This discussion should be closed like https://github.com/OpenModelica/OpenModelica/discussions/9547.
